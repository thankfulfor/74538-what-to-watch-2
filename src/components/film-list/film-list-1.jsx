import * as React from 'react';
import PropTypes from 'prop-types';
import FilmCard from '../film-card/film-card.jsx';
import withActiveItem from '../../hoc/with-active-item/with-active-item.jsx';

const FilmCardWrapped = withActiveItem(FilmCard);

export const FilmList = (props) => {
  const {films} = props;

  return (
    <div className="catalog__movies-list">
      {films.map((film, i) => <FilmCardWrapped key={`${film.name}-${i}`} film={film} />)}
    </div>
  );
};

FilmList.propTypes = {
  films: PropTypes.array.isRequired,
};
