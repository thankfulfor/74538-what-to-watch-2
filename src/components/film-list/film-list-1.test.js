import React from 'react';
import renderer from 'react-test-renderer';
import {FilmList} from './film-list-1.jsx';
import {BrowserRouter} from 'react-router-dom';

const films = [{
  previewImage: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
  genre: `drama`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv`
}];

it(`FilmList корректно рендерится после перезапуска`, () => {
  const tree = renderer
    .create(<BrowserRouter><FilmList films={films} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
