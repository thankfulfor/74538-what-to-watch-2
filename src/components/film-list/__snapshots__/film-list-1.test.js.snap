// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList корректно рендерится после перезапуска 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <FilmList
    films={
      Array [
        Object {
          "genre": "drama",
          "name": "We need to talk about Kevin",
          "previewImage": "img/we-need-to-talk-about-kevin.jpg",
          "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv",
        },
      ]
    }
  />
</Router>
`;
